src/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    /// <summary>
+    /// Wrapper around an output gss_buffer_desc*
+    /// </summary>
+    internal sealed class SafeGssBufferHandle : SafeHandle
+    {
+        private SafeGssBufferHandle() : base(IntPtr.Zero, true)
+        {
+        }
+
+        public override bool IsInvalid
+        {
+            get { return handle == IntPtr.Zero; }
+        }
+
+        internal void Copy(byte[] destination, int offset)
+        {
+            Debug.Assert(destination != null, "destination buffer cannot be null");
+            Debug.Assert(offset <= destination.Length , "offset in the destination buffer must be valid");
```

Comment:
Do we know how big the represented buffer is so that we can validate against the actual length as well?  And I assume that the inputs to this method are constructed by us internally and validated such that release code with invalid input could never result in invalid inputs here?